Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocColumnRendererArrayBuilderTest.cs

```diff
@@ -57,7 +57,32 @@ public void GetColumnRenderers ()
     }
 
     [Test]
-    public void GetColumnRenderers_WithBocValidationErrorIndicatorColumnDefinition_AlwaysVisibility_WithoutFailures_ContainsNullColumnRenderer ()
+    public void CreateColumnRenderers_WithNotVisibleColumn_CreatesNullColumnRenderer ()
+    {
+      var notVisibleColumn = new StubColumnDefinition { IsVisible = false };
+      var firstColumn = new StubColumnDefinition();
+      var thirdColumn = new StubColumnDefinition();
+
+      var builder = new BocColumnRendererArrayBuilder(CreateColumnCollection(firstColumn, notVisibleColumn, thirdColumn), _serviceLocator);
+
+      var bocColumnRenderers = builder.CreateColumnRenderers();
+
+      Assert.That(bocColumnRenderers.Length, Is.EqualTo(3));
+      Assert.That(PrivateInvoke.GetNonPublicField(bocColumnRenderers[1], "_columnRenderer"), Is.TypeOf(typeof(NullColumnRenderer)));
+      Assert.That(bocColumnRenderers[1].ColumnDefinition, Is.SameAs(notVisibleColumn));
+      Assert.That(bocColumnRenderers[1].ColumnIndex, Is.EqualTo(1));
```

Comment:
```suggestion
      Assert.That(bocColumnRenderers[1].ColumnIndex, Is.EqualTo(1));
      Assert.That(bocColumnRenderers[1].VisibleColumnIndex, Is.EqualTo(-1));
```

Reply:
Asserting the second property is also important because we want to document that the visible column index skips an item.